libraries/AP_InertialSensor/AP_InertialSensor_SCH16T.cpp

```diff
@@ -0,0 +1,448 @@
+/*
+ * This file is free software: you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License as published by the
+ * Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This file is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include <AP_Math/AP_Math.h>
+
+#include "AP_InertialSensor_SCH16T.h"
+// #include <GCS_MAVLink/GCS.h>
+
+#if defined(HAL_GPIO_PIN_nSPI6_RESET_EXTERNAL1)
+#include <hal.h>
+#endif
+
+static constexpr uint32_t SPI_SPEED = 2 * 1000 * 1000;  // 2 MHz SPI serial interface
+static constexpr uint32_t SAMPLE_INTERVAL_US = 678;     // 1500 Hz -- decimation factor 8, F_PRIM/16, 1.475 kHz
+static constexpr uint16_t EOI = (1 << 1);               // End of Initialization
+static constexpr uint16_t EN_SENSOR = (1 << 0);         // Enable RATE and ACC measurement
+static constexpr uint16_t DRY_DRV_EN = (1 << 5);        // Enables Data ready function
+static constexpr uint16_t FILTER_68HZ = (0x0000);       // 68 Hz default filter
+static constexpr uint16_t RATE_300DPS_1475HZ = 0b0001001011011011; // Gyro XYZ range 300 deg/s @ 1475Hz
```

Comment:
This is too low for a flying vehicle I think, are there not any options for higher rates?

Reply:
Not on this version no, but they will be releasing higher range versions (soon)

The datasheet also isn't publicly available but these are the highlights from the top
![image](https://github.com/ArduPilot/ardupilot/assets/37091262/495c2a39-4bc1-4bce-a295-17a7dfbae091)